WordPress/Classes/Services/PostHelper.m

```diff
@@ -238,7 +248,12 @@ + (NSArray *)mergePosts:(NSArray <RemotePost *> *)remotePosts
 {
     NSMutableArray *posts = [NSMutableArray arrayWithCapacity:remotePosts.count];
     for (RemotePost *remotePost in remotePosts) {
-        AbstractPost *post = [blog lookupPostWithID:remotePost.postID inContext:context];
+        AbstractPost *post;
+        if (remotePost.foreignID != nil) {
+            post = [blog lookupPostWithForeignID:remotePost.foreignID inContext:context];
```

Comment:
I'm able to reproduce the issue with the following steps:

https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/01b2a4bd-c453-44e8-9a70-25dc66973228